src/d3a/d3a_core/util.py

```diff
@@ -572,3 +573,41 @@ def wrapper(*args, **kwargs):
 def get_simulation_queue_name():
     listen_to_cn = os.environ.get("LISTEN_TO_CANARY_NETWORK_REDIS_QUEUE", "no") == "yes"
     return "canary_network" if listen_to_cn else "d3a"
+
+
+class ExternalTickTimer:
+
+    def __init__(self, ticks_per_slot):
+        self._last_dispatched_tick = 0
+        self.ticks_per_slot = ticks_per_slot
+
+    @property
+    def _dispatch_tick_frequency(self) -> int:
+        return int(
+            self.ticks_per_slot *
+            (DISPATCH_EVENT_TICK_FREQUENCY_PERCENT / 100)
+        )
+
+    def reset_event_tick_counter(self):
+        self._last_dispatched_tick = 0
+
+    def _update_current_tick(self, current_tick):
+        self._last_dispatched_tick = current_tick
+
+    def is_it_time_for_external_tick(self, current_tick_in_slot) -> bool:
+        if current_tick_in_slot - self._last_dispatched_tick >= self._dispatch_tick_frequency:
+            self._update_current_tick(current_tick_in_slot)
+            return True
+        else:
+            return False
+
+
+def get_market_maker_rate_from_global_setting(next_market):
+    if isinstance(GlobalConfig.market_maker_rate, dict):
+        if next_market is None:
+            return GlobalConfig.market_maker_rate[next(iter(GlobalConfig.market_maker_rate))]
+        else:
+            return find_object_of_same_weekday_and_time(GlobalConfig.market_maker_rate,
+                                                        next_market.time_slot)
+    else:
+        return GlobalConfig.market_maker_rate
```

Comment:
I think that this does not cover the case of islanded mode. I recall from a discussion with Chris, that he was interested in the feed-in-tariff, which is not exactly the market maker rate (we were supposed to report a None value if the market maker is not an infinite bus or commercial producer, but a simple market maker that sets a market maker rate and does not place offers and bids)

Reply:
Actually I forgot about the `feed_in_tariff_rate`. Will add it and of course. Thanks for reminding.
The above was just a refactoring of getting the market_maker as it was called in the load, the pv and now also in the global_objects.

Reply:
Actually, no there is a ticket D3ASIM-3350 for that. So, lets add this later.